pkg/alertmanager/alertstore/config.go

```diff
@@ -51,31 +59,79 @@ func (cfg *Config) Validate() error {
 	return nil
 }
 
-// NewAlertStore returns a new rule storage backend poller and store
-func NewAlertStore(cfg Config) (AlertStore, error) {
-	switch cfg.Type {
-	case "configdb":
+// IsDefaults returns true if the storage options have not been set.
+func (cfg *LegacyConfig) IsDefaults() bool {
+	return cfg.Type == ConfigDB && cfg.ConfigDB.ConfigsAPIURL.URL == nil
+}
+
+// NewLegacyAlertStore returns a new alertmanager storage backend poller and store
+func NewLegacyAlertStore(cfg LegacyConfig, logger log.Logger) (AlertStore, error) {
+	if cfg.Type == "configdb" {
 		c, err := client.New(cfg.ConfigDB)
 		if err != nil {
 			return nil, err
 		}
 		return configdb.NewStore(c), nil
+	}
+
+	if cfg.Type == "local" {
+		return local.NewStore(cfg.Local)
+	}
+
+	// Create the object store client.
+	var client chunk.ObjectClient
+	var err error
+	switch cfg.Type {
 	case "azure":
-		return newObjAlertStore(azure.NewBlobStorage(&cfg.Azure))
+		client, err = azure.NewBlobStorage(&cfg.Azure)
 	case "gcs":
-		return newObjAlertStore(gcp.NewGCSObjectClient(context.Background(), cfg.GCS))
+		client, err = gcp.NewGCSObjectClient(context.Background(), cfg.GCS)
 	case "s3":
-		return newObjAlertStore(aws.NewS3ObjectClient(cfg.S3))
-	case "local":
-		return local.NewStore(cfg.Local)
+		client, err = aws.NewS3ObjectClient(cfg.S3)
 	default:
 		return nil, fmt.Errorf("unrecognized alertmanager storage backend %v, choose one of: azure, configdb, gcs, local, s3", cfg.Type)
 	}
+	if err != nil {
+		return nil, err
+	}
+
+	return objectclient.NewAlertStore(client, logger), nil
 }
 
-func newObjAlertStore(client chunk.ObjectClient, err error) (AlertStore, error) {
+// Config configures a the alertmanager storage backend.
+type Config struct {
+	bucket.Config `yaml:",inline"`
+	ConfigDB      client.Config `yaml:"configdb"`
+}
+
+// RegisterFlags registers the backend storage config.
+func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
+	prefix := "alertmanager-storage."
+
+	cfg.ExtraBackends = []string{ConfigDB}
+	cfg.ConfigDB.RegisterFlagsWithPrefix(prefix, f)
+	cfg.RegisterFlagsWithPrefix(prefix, f)
+}
+
+// NewAlertStore returns a alertmanager store backend client based on the provided cfg.
+func NewAlertStore(ctx context.Context, cfg Config, cfgProvider bucket.TenantConfigProvider, logger log.Logger, reg prometheus.Registerer) (AlertStore, error) {
+	if cfg.Backend == ConfigDB {
+		c, err := client.New(cfg.ConfigDB)
+		if err != nil {
+			return nil, err
+		}
+		return configdb.NewStore(c), nil
+	}
+
+	bucketClient, err := bucket.NewClient(ctx, cfg.Config, "alertmanager-storage", logger, reg)
 	if err != nil {
 		return nil, err
 	}
-	return objectclient.NewAlertStore(client), nil
+
+	store := bucketclient.NewBucketAlertStore(bucketClient, cfgProvider, logger)
+	if err != nil {
```

Comment:
`NewBucketAlertStore` doesn't return an error - no need to check `err`.

Can just `return  bucketclient.NewBucketAlertStore(...), nil`